dove/src/stdout.rs

```diff
@@ -0,0 +1,213 @@
+        .unwrap()
+        .as_mut()
+        .print(text)
+}
+
+/// Print output with a newline.
+pub fn println(text: &str) -> Result<(), Error> {
+    print(text)?;
+    print("\n")
+}
+
+/// print output in green color
+pub fn print_good(text: &str) -> Result<(), Error> {
+    print_with_color(text, ColorSpec::new().set_fg(Some(Color::Green)))
+}
+
+/// print output in red color
+pub fn print_error(text: &str) -> Result<(), Error> {
+    print_with_color(text, ColorSpec::new().set_fg(Some(Color::Red)))
+}
+
+/// print output in yellow color
+pub fn print_warning(text: &str) -> Result<(), Error> {
+    print_with_color(text, ColorSpec::new().set_fg(Some(Color::Yellow)))
+}
+
+/// print output with color settings
+fn print_with_color(text: &str, spec: &mut ColorSpec) -> Result<(), Error> {
+    let mut buffer = BufferWriter::stdout(get_stdout_color()).buffer();
+    buffer.set_color(spec)?;
+    write!(&mut buffer, "{}", text)?;
+    buffer.reset()?;
+    print(&String::from_utf8_lossy(buffer.as_slice()))
+}
+
+/// Prints output.
+#[macro_export]
+macro_rules! stdout {
+    ($($args:tt)*) => {
+        $crate::stdout::print(&format!($($args)*)).unwrap();
+    };
+}
+
+/// Prints output with a newline.
+#[macro_export]
+macro_rules! stdoutln {
+    () => ($crate::stdout::print!("\n").unwrap());
+    ($($args:tt)*) => {
+        $crate::stdout::println(&format!($($args)*)).unwrap();
+    }
+}
+
+/// Set the stream / buffer to write
+pub fn set_buffer<Out>(stdout: Out) -> Result<(), Error>
+where
+    Out: TBufWrite + Send,
+    Out: 'static,
+{
+    let flag = get_argv_flag_color(&stdout);
+
+    match STDOUT_COLOR.get() {
+        Some(mt) => {
+            let mut buff = mt.lock().unwrap();
+            *buff = Box::new(flag);
+            Ok(())
+        }
+        None => STDOUT_COLOR
+            .set(Mutex::new(Box::new(flag)))
+            .map_or_else(|_| Err(anyhow!("Failed to set STDOUT_COLOR")), |_| Ok(())),
+    }?;
+
+    match STDOUT_STREAM.get() {
+        Some(mt) => {
+            let mut buff = mt.lock().unwrap();
+            *buff = Box::new(stdout);
+            Ok(())
+        }
+        None => STDOUT_STREAM
+            .set(Mutex::new(Box::new(stdout)))
+            .map_or_else(|_| Err(anyhow!("Failed to set buffer output")), |_| Ok(())),
+    }
+}
+
+/// Get data from the buffer if a string was used
+pub fn get_buffer_value() -> Option<String> {
+    STDOUT_STREAM
+        .get()
+        .and_then(|mt| mt.lock().map_or(None, |bx| bx.value()))
+}
+
+/// get marker to highlight
+fn get_stdout_color() -> ColorChoice {
+    match STDOUT_COLOR.get() {
+        Some(mt) => *(mt.lock().unwrap()).clone(),
+        None => ColorChoice::Auto,
+    }
+}
+
+/// get flag --color=.. auto|always|never|ansi
+fn get_argv_flag_color<Out>(stdout: &Out) -> ColorChoice
+where
+    Out: TBufWrite + Send,
+    Out: 'static,
+{
+    let flag = std::env::args()
+        .find_map(|it| {
+            if it.to_lowercase().contains("--color=") {
```

Comment:
Can't we use `structopt` for that? There are at least two downsides of the given code:
1. It does not handle `--color always` syntax
2. It does not provide any help messages for the user.

Btw, why there is no `--color=never` or `--color=none` variants? `ansi` seems counter-intuitive.

Reply:
This is taken from the documentation for termcolor.
https://docs.rs/termcolor/1.1.2/termcolor/#detecting-presence-of-a-terminal

>Btw, why there is no --color=never or --color=none variants? ansi seems counter-intuitive.

Why not? Everything except ansi, auto, always will be never

Reply:
> It does not handle --color always syntax
Yes, you're right. It needs to be processed in **structopt**. And catch here. 
It's in a different PR

Reply:
> Why not? Everything except ansi, auto, always will be never

Okay, it was not clear from the code. Will you create an issue to address this?